src/content/blog/astro-4120.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: "Astro 4.12: Server Islands"
+description: Astro 4.12 is now available! This release includes a first look at Server Islands, improvements to pagination and syntax highlighting, and more.
+publishDate: "July 18, 2024"
+authors:
+  - erika
+  - matthew
+coverImage: "/src/content/blog/_images/astro-4120/header_astro_4_12.webp"
+socialImage: "/src/content/blog/_images/astro-4120/og_image_4_12.webp"
+lang: "en"
+---
+
+import BlogSummary from "/src/components/BlogSummary.astro";
+
+<BlogSummary>
+  Astro 4.12 is out now! This release includes a first look at [Server Islands](https://astro.build/blog/future-of-astro-server-islands/), as well as improvements to pagination and syntax highlighting, and more.
+</BlogSummary>
+
+This release includes the following highlights:
+
+- [**Experimental: Server Islands**](#experimental-server-islands)
+- [**Added `first` and `last` URLs to pagination data**](#added-first-and-last-urls-to-pagination-data)
+- [**Added support for the Shiki's `defaultColor` option**](#added-support-for-the-shikis-defaultcolor-option)
+- [**New `inferRemoteSize` function**](#new-inferremotesize-function)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Server Islands
+
+TODO: Matthew Please Write This Part
+
+## Added `first` and `last` URLs to pagination data
+
+The [paginate() helper](https://docs.astro.build/en/reference/api-reference/#paginate) now returns the `first` and `last` properties in the pagination data. These properties contain the first and last page urls, respectively. Thanks to [@tsawada](https://github.com/tsawada) for contributing this feature!
+
+## Added support for the Shiki's `defaultColor` option
+
+The `defaultColor` option allows you to override the values of a theme's inline style, adding only CSS variables to give you more flexibility in applying multiple color themes. This option can be configured in your Shiki config to apply throughout your site, or passed to Astro's built-in `<Code>` component to style an individual code block. Thank you to [@madcampos](https://github.com/madcampos) for contributing this!
```

Comment:
``````suggestion
The `defaultColor` option allows you to override the values of a theme's inline style, adding only CSS variables to give you more flexibility in applying multiple color themes.

This option can be configured in your Shiki config to apply throughout your site, or passed to Astro's built-in `<Code>` component to style an individual code block. Thank you to [@madcampos](https://github.com/madcampos) for contributing this!

```js title="astro.config.mjs"
import { defineConfig } from 'astro/config';
export default defineConfig({
  markdown: {
    shikiConfig: {
      themes: {
        light: 'github-light',
        dark: 'github-dark',
      },
      defaultColor: false,
    },
  },
});
```

```astro
---
import { Code } from 'astro:components';
---
<Code code={`const useMyColors = true`} lang="js" defaultColor={false} />
```
``````
Totally not necessary, just throwing these here in case you liked having the examples visually to break up text! People like to skim for code!

Reply:
I *do* like it, however I've deliberately kept the sections that are not "Server Islands" small-ish so for marketing-purposes, so not sure!

Reply:
Entirely your call! That does make sense! You might have a better sense after the Server Islands part is added whether it feels right.